src/Uno.UI.RemoteControl.Host/RemoteControlServer.cs

```diff
@@ -12,55 +12,103 @@
 using Uno.UI.RemoteControl.Messages;
 using System.Runtime.Loader;
 using Microsoft.Extensions.Configuration;
+using System.Reflection;
+using System.Runtime.Remoting.Messaging;
 
 namespace Uno.UI.RemoteControl.Host
 {
 	internal class RemoteControlServer : IRemoteControlServer, IDisposable
 	{
-		private readonly Dictionary<string, IServerProcessor> _processors = new Dictionary<string, IServerProcessor>();
+		private readonly Dictionary<string, IServerProcessor> _processors = new();
 
 		private System.Reflection.Assembly? _loopAssembly;
 		private WebSocket? _socket;
+		private string? _appInstanceId;
 		private readonly IConfiguration _configuration;
-		private readonly AssemblyLoadContext _loadContext;
+		private readonly static Dictionary<string, AssemblyLoadContext> _loadContexts = new();
 
 		public RemoteControlServer(IConfiguration configuration)
 		{
 			_configuration = configuration;
-			_loadContext = new AssemblyLoadContext(null, isCollectible: true);
-			_loadContext.Unloading += (e) =>
+
+			if (this.Log().IsEnabled(LogLevel.Debug))
+			{
+				this.Log().LogDebug("Starting RemoteControlServer");
+			}
+		}
+
+		string IRemoteControlServer.GetServerConfiguration(string key)
+			=> _configuration[key] ?? "";
+
+		private void EnsureLoadContextForApplicationId(string applicationId)
+		{
+			if (_loadContexts.ContainsKey(applicationId))
+			{
+				return;
+			}
+
+			var loadContext = new AssemblyLoadContext(null, isCollectible: true);
```

Comment:
We might as well set the name of the AssemblyLoadContext to be the applicationId